get-started/debug-and-profile.md

```diff
@@ -1,8 +1,174 @@
 
-* how to debug tidb
-* how to write benchmarks
-* how to profile
+## Use delve for debugging
 
+[Delve](https://github.com/go-delve/delve) is a debugger for the Go programming language. It provides GDB-like command-line debugging experience and is much more Go native.
+
+### Install delve
+
+To install delve, you can refer to the [installation guide](https://github.com/go-delve/delve/tree/master/Documentation/installation). After the installation, you will have an executable file named `dlv` in `$GOPATH/bin` or `$HOME/go/bin` depending on how your environment variables are set, and you can then type the following command to verify the installation.
+
+```
+$ dlv version
+Delve Debugger
+Version: 1.5.0
+Build: $Id: ca5318932770ca063fc9885b4764c30bfaf8a199 $
+```
+
+### Attaching delve to a running TiDB process
+
+As mentioned in [Get the code, build and run](build-tidb-from-source.md), once you get the TiDB server running, you can try to attach the delve debugger.
+
+For example, you can typically build and run a standalone TiDB server by
+
+```
+$ make server
+$ ./bin/tidb-server
+```
+
+in the root directory of the source code.
+
+You can then start a new shell and use `ps` or `pgrep` to find the PID of the tidb server process you just started.
+
+```
+$ pgrep tidb-server
+1394942
+```
+
+If there are multiple PIDs listed, it might be that you have multiple tidb servers running, and you should figure out the PID of the tidb server you are planning to debug with using commands like `ps $PID` where `$PID` is the PID you are trying to know more about.
+
+```
+$ ps 1394942
+    PID TTY      STAT   TIME COMMAND
+1394942 pts/11   SNl    0:02 ./bin/tidb-server
+```
+
+Once you got the PID, you can then attach delve onto it by
+
+```
+$ dlv attach 1394942
+```
+
+You might encounter error messages like
+
+```
+$ dlv attach 1394942
+Could not attach to pid 1394942: this could be caused by a kernel security setting, try writing "0" to /proc/sys/kernel/yama/ptrace_scope
+```
+
+You should follow the instruction and execute the following command as the root user to override the kernel security setting.
+
+```
+# echo 0 > /proc/sys/kernel/yama/ptrace_scope
+```
+
+You should then retry attaching and it should work.
+
+Entering the debugging interface, you will find it familiar if you’ve worked with GDB. It is a interactive dialogue that allows you to interact with the execution of the tidb server attached on. You should type `help` into the dialogue and read the help messages.
+
+### Using delve for debugging
+
+After attaching delve to the running TiDB server process, you can now set breakpoints, and the TiDB server will pause the execution at the breakpoints you specified.
+
+To create a breakpoint, you can write
+
+```
+break [name] <linespec>
```

Comment:
Is <linespec> a line number in the code? Could we provide an example here to avoid confusion?

Reply:
sure.